pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -578,24 +850,21 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 		return nil, statusUnschedulable(logger, err.Error())
 	}
 	logger.V(5).Info("pod resource claims", "pod", klog.KObj(pod), "resourceclaims", klog.KObjSlice(claims))
+
 	// If the pod does not reference any claim,
 	// DynamicResources Filter has nothing to do with the Pod.
 	if len(claims) == 0 {
 		return nil, framework.NewStatus(framework.Skip)
 	}
 
-	// Fetch s.podSchedulingState.schedulingCtx, it's going to be needed when checking claims.
+	// Fetch PodSchedulingContext, it's going to be needed when checking claims.
 	if err := s.podSchedulingState.init(ctx, pod, pl.podSchedulingContextLister); err != nil {
 		return nil, statusError(logger, err)
 	}
 
 	s.informationsForClaim = make([]informationForClaim, len(claims))
+	needResourceInformation := false
 	for index, claim := range claims {
-		if claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeImmediate &&
-			claim.Status.Allocation == nil {
-			// This will get resolved by the resource driver.
-			return nil, statusUnschedulable(logger, "unallocated immediate resourceclaim", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
-		}
```

Comment:
This had to be moved because at this point we don't have the class for the claim yet and thus don't know whether we really depend on a DRA resource driver in the control plane.